tests/area/test_markets.py

```diff
@@ -0,0 +1,85 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+
+from unittest.mock import Mock
+
+import pytest
+from d3a_interface.constants_limits import ConstSettings, GlobalConfig
+from pendulum import duration
+
+from d3a.d3a_core.device_registry import DeviceRegistry
+from d3a.models.area import Area
+from d3a.models.strategy.storage import StorageStrategy
+
+
+class TestMarketRotation:
+
+    @pytest.fixture
+    def area_fixture(self):
+        DeviceRegistry.REGISTRY = {
+            "General Load": (33, 35),
+        }
+        config = Mock()
+        config.slot_length = duration(minutes=15)
+        config.tick_length = duration(seconds=15)
+        config.ticks_per_slot = 60
+        config.start_date = GlobalConfig.start_date
+        config.grid_fee_type = ConstSettings.IAASettings.GRID_FEE_TYPE
+        config.end_date = GlobalConfig.start_date + duration(days=1)
+        config.market_count = 5
+        child = Area(name="test_market_area", config=config, strategy=StorageStrategy())
+        area = Area(name="parent_area", children=[child], config=config)
+        yield area
+
+        DeviceRegistry.REGISTRY = {}
+
+    def test_cycle_markets(self, area_fixture):
+        ConstSettings.BalancingSettings.ENABLE_BALANCING_MARKET = True
+        area_fixture.activate()
+        assert len(area_fixture.all_markets) == 5
```

Comment:
Because the market count is set to 5:
https://github.com/gridsingularity/d3a/pull/1213/files/967e4d58532940a78e334daf10e935b8af11a33f#diff-1a3117c0a30c4f92672a17fce97e08afb57e80744eb491247394ac5e43fed03fR45